appengine-vmruntime/vmruntime/middleware.py

```diff
@@ -235,3 +236,22 @@ def health_check_intercept_wrapper(request):
             return app
 
     return health_check_intercept_wrapper
+
+
+def callback_middleware(app):
+    """Calls the request-endcallback that the app may have set
+
+    Args:
+        app: The WSGI app to wrap.
+
+    Returns:
+        The wrapped app, also a WSGI app.
+    """
+    def callback_wrapper(environ, start_response):
+        """Call the WSGI app and then invoke the request-end callbacks."""
+        try:
+            return app(environ, start_response)
+        finally:
+            callback.InvokeCallbacks()
```

Comment:
Hmm, I'm not sure if there might be any strange consequences to doing it with a finally:. I was imagining you would simply execute the callback as the final step before returning the response -- for instance,

    response = app(environ, start_response)
    callback.InvokeCallbacks()
    return response

I'm not sure about the pros and cons yet, so I'm okay with whichever method you choose. LGTM.